receiver/awscontainerinsightreceiver/internal/k8sapiserver/k8sapiserver.go

```diff
@@ -537,6 +538,44 @@ func (k *K8sAPIServer) getAcceleratorCountMetrics(clusterName, timestampNs strin
 	return metrics
 }
 
+func (k *K8sAPIServer) getHyperPodResiliencyMetrics(clusterName, timestampNs string) []pmetric.Metrics {
+	var metrics []pmetric.Metrics
+	for nodeName, labels := range k.leaderElection.nodeClient.NodeToLabelsMap() {
+		if isHyperPodNode(nodeName) {
+			fields := map[string]any{}
+			attributes := map[string]string{
+				ci.ClusterNameKey: clusterName,
+				ci.MetricType:     ci.TypeHyperPodNode,
+				ci.Timestamp:      timestampNs,
+				ci.Version:        "0",
+			}
+
+			isUnschedulable := 0
+			for _, condition := range []k8sutil.HyperPodConditionType{
+				k8sutil.UnschedulablePendingReplacement,
+				k8sutil.UnschedulablePendingReboot,
+				k8sutil.Schedulable,
+				k8sutil.SchedulablePreferred,
+			} {
+				if status, ok := isLabelSet(condition, labels, k8sclient.SageMakerNodeHealthStatusSC); ok {
+					fields[ci.MetricName(ci.TypeHyperPodNode, ci.ConditionToMetricName[condition.String()])] = status
+
+					if status == 1 && (condition == k8sutil.UnschedulablePendingReplacement || condition == k8sutil.UnschedulablePendingReboot) {
+						isUnschedulable = 1
+					}
+				}
+			}
+			fields[ci.MetricName(ci.TypeHyperPodNode, ci.ConditionToMetricName[k8sutil.Unschedulable.String()])] = isUnschedulable
+			fields[ci.MetricName(ci.TypeHyperPodNode, ci.ConditionToMetricName[k8sutil.Unknown.String()])] = isLabelUnknown(labels, k8sclient.SageMakerNodeHealthStatusSC)
+			attributes[ci.InstanceID] = strings.TrimLeft(nodeName, "hyperod-")
```

Comment:
Typo here. `hyperpod-`

Reply:
>  SA1024: cutset contains duplicate characters (staticcheck)
> A cutset is treated as a set of characters to remove from a
string


https://pkg.go.dev/honnef.co/go/tools/staticcheck

Reply:
Thinking about this again, we shouldn't trim this, lets find out the rule that breaks without a valid instance-id and fix that instead.